service/header/core_listener.go

```diff
@@ -0,0 +1,113 @@
+package header
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/tendermint/tendermint/types"
+)
+
+// CoreListener TODO @renaynay: document
+type CoreListener struct {
+	ex     *CoreExchange
```

Comment:
From CoreExchnage you need only a fetcher, why not to rcv it directly?

Reply:
* Lesser coupling
* Eliminates the need to provide CoreExchange as a ptr

Reply:
kk

Reply:
actually no, i need exchange's helper functions to generate ExtendedHeader

Reply:
Maybe then extract this helper func?

Reply:
It is gross when two independent structs access each others private funcs

Reply:
okey

Reply:
@Wondertan `generateExtendedHeader` is very much tied to fetcher since it accesses a couple different methods on fetcher to construct the ExtendedHeader

Either I can export this method on `CoreExchange` or I can make the method on fetcher instead of `CoreExchange`. Wdyt?

Reply:
>  I can make the method on fetcher instead of CoreExchange

This will be a dep cycle.

Reply:
> generateExtendedHeader is very much tied to fetcher

Do you mean tied to CoreExchange?

Reply:
tied to CoreExchange I mean. Should I just export that method?